test/test_postman.py

```diff
@@ -48,26 +65,30 @@ def test_init_ganache():
 
 @pytest.mark.web3_deploy
 def test_deploy_l1_contracts():
-    """Deploys Ethereum contracts in the Ganache instance, including the L1L2Example and MockStarknetMessaging contracts"""
+    """Deploys Ethereum contracts in the Hardhat testnet instance, including the L1L2Example and MockStarknetMessaging contracts"""
 
-    global MESSAGING_CONTRACT_ADDRESS # pylint: disable=global-statement
-    global L1L2_EXAMPLE_CONTRACT_ADDRESS # pylint: disable=global-statement
+    global MESSAGING_CONTRACT_DEFINITION # pylint: disable=global-statement
+    global L1L2_EXAMPLE_CONTRACT_DEFINITION # pylint: disable=global-statement
+    global WEB3 # pylint: disable=global-statement disable=global-variable-not-assigned
 
-    args = "cd test && truffle migrate && cd .."
-    subprocess.run(args, shell=True, encoding="utf-8", check=False, capture_output=True)
-    messaging_contract = json.loads(load_file_content(STARKNET_MESSAGING_PATH))
-    l1l2_example_contract = json.loads(load_file_content(L1L2_EXAMPLE_ETH_PATH))
+    install_solc(Version('0.6.12'))
+    set_solc_version(Version('0.6.12'))
+    install_solc_pragma("pragma solidity ^0.6.12;")
+    messaging_contract = load_file_content(STARKNET_MESSAGING_PATH)
+    l1l2_example_contract = load_file_content(L1L2_EXAMPLE_ETH_PATH)
 
-    MESSAGING_CONTRACT_ADDRESS = messaging_contract["networks"]["32"]["address"]
-    L1L2_EXAMPLE_CONTRACT_ADDRESS = l1l2_example_contract["networks"]["32"]["address"]
+    messaging_contract_interface = web3_compile(messaging_contract, os.path.abspath(SOLIDITY_CONTRACTS_PATH))
```

Comment:
can't compilation artifacts be expected to exist? `npx hardhat compile` step could be added before or after `npx hardhat starknet-compile` in `setup_example.sh`: https://github.com/Shard-Labs/starknet-devnet/blob/master/scripts/setup_example.sh#L23
If that could be achieved, then the pysolc dependency (dev or not) wouldn't be needed.

Reply:
You are right, that's a much simpler way. Changed do use `npx hardhat compile`, thanks